README.md

```diff
@@ -1,14 +1,38 @@
-# The Noir Programming Language Syntax Highlighter
+# Noir Language Support for Visual Studio Code
 
-## How to publish?
+Visual Studio Code extension for working with the [Noir programming language](https://noir-lang.org/).
 
-https://code.visualstudio.com/api/working-with-extensions/publishing-extension
+## Features
 
-## How it is generated?
+This extension helps developers write, understand, and improve Noir code by providing:
 
-https://code.visualstudio.com/api/get-started/your-first-extension
+* Syntax highlighting
+* Compile errors and warnings on file save
+* Useful snippets for common code patterns
 
-```
-yo code
-> New Language Support
-```
\ No newline at end of file
+## Requirements
+
+For Language Server Protocol support, we require `nargo` v0.7.1+ to be in your `PATH`. You can check this by running `nargo --version` in your terminal and you can check for the `lsp` command by running `nargo --help`.
+
+If you can't put `nargo` in your `PATH`, you can set an absolute path in the `Nargo Path` [setting](#settings).
+
+## Settings
+
+* __Noir: Enable LSP__ _(noir.enableLSP)_ - If checked, the extension will launch the Language Server via `nargo lsp` and communicate with it.
+* __Noir: Nargo Flags__ _(noir.nargoFlags)_ - Additional flags may be specified if you require them to be added when the extension calls `nargo lsp`.
+* __Noir: Nargo Path__ _(noir.nargoPath)_ - An absolute path to a Nargo binary with the `lsp` command. This may be useful if Nargo is not within the `PATH` of your editor.
+* __Noir > Trace: Server__ _(noir.trace.server)_ - Setting this to `"messages"` or `"verbose"` will log LSP messages between the Client and Server. Useful for debugging.
+
+## Changelog
+
+You can find a full list of changes at https://github.com/noir-lang/vscode-noir/blob/master/CHANGELOG.md
+
+## Working on project
+
+The project provides a few useful commands via npm scripts:
+
+* `npm run package` - Builds the project and packages it into a `.vsix` file which can be manually installed for testing.
+* `npm run esbuild` - Builds the project with esbuild to output `out/extension.js` and `out/extension.js.map`.
+* `npm run test-compile` - Check types with TypeScript. Useful since esbuild doesn't typecheck.
+* `npm run format` - Formats all the code in the repository.
+* `npm run check-format` - Checks the formatting in the repository.
```

Comment:
Should this section be removed ?

Reply:
I think it's okay to have this in the readme currently for people that need to work on the extension. If we grow this section, it'll be split into a contributor doc.